File: src/dere_discord/paths.py
"""Shared filesystem helpers for dere-discord."""

from __future__ import annotations

from dere_shared.paths import get_config_dir

__all__ = ["get_config_dir", "format_project_path"]


def format_project_path(
    *,
    guild_id: int | None,
    channel_id: int,
    user_id: int | None = None,
) -> str:
    """Build a project path string understood by the daemon."""

    if guild_id is None:
        return f"discord://dm/{user_id or channel_id}"
    return f"discord://guild/{guild_id}/channel/{channel_id}"

File: packages/discord/src/paths.ts
import { dirname } from "node:path";
import { getConfigPath } from "@dere/shared-config";

export function getConfigDir(): string {
  return dirname(getConfigPath());
}

export function formatProjectPath(args: {
  guildId: string | null;
  channelId: string;
  userId?: string | null;
}): string {
  const { guildId, channelId, userId } = args;
  if (guildId === null) {
    return `discord://dm/${userId ?? channelId}`;
  }
  return `discord://guild/${guildId}/channel/${channelId}`;
}
